util/util.go

```diff
@@ -355,3 +356,20 @@ func Index(slice []string, elem string) int {
 	}
 	return -1
 }
+
+//ReadJSON reads JSON from http request
+func ReadJSON(r *http.Request) (map[string]interface{}, error) {
+	var data interface{}
+	var reqData bytes.Buffer
+	tee := io.TeeReader(r.Body, &reqData)
+	decoder := json.NewDecoder(tee)
+	r.Body = ioutil.NopCloser(&reqData)
```

Comment:
looks fishy: we're modifying the requests internal state in a utility function. instead I'd rather store (a copy of) the `Body` in the `context`. as a result you wouldn't need to provide `GetHttpRequestBody` in `goext`

Reply:
You already keep request inside context, lets not duplicate this buffer. Reading from r.Body is changing internal state since r.Body is a stream which is consumed upon read. This function gives guarantee that r.Body will be readable multiple times.